internal/cmd/extensions/list.go

```diff
@@ -43,53 +43,68 @@ func ListCmd(options *clioptions.CLIOptions) *cobra.Command {
 			}
 
 			extensibilityClient := New(client)
-			extensions, err := extensibilityClient.List(cmd.Context(), restConfig.CompanyID)
+			extensions, err := extensibilityClient.List(cmd.Context(), restConfig.CompanyID, options.ResolveExtensionsDetails)
 			cobra.CheckErr(err)
 
-			printExtensionsList(extensions, options.Printer(clioptions.DisableWrapLines(true)))
+			printExtensionsList(extensions, options.Printer(clioptions.DisableWrapLines(true)), options.ResolveExtensionsDetails)
 			return nil
 		},
 	}
+
+	addResolveDetailsFlag(options, cmd)
+	return cmd
+}
+
+func addResolveDetailsFlag(options *clioptions.CLIOptions, cmd *cobra.Command) {
+	flags := cmd.Flags()
+	flags.BoolVar(&options.ResolveExtensionsDetails, "resolve-details", false, "Retrieve also menu and category info")
 }
 
-func printExtensionsList(extensions []*extensibility.ExtensionInfo, p printer.IPrinter) {
-	p.Keys("ID", "Name", "Entry", "Destination", "Menu (id) / Category (id)", "Description")
+func printExtensionsList(extensions []*extensibility.ExtensionInfo, p printer.IPrinter, resolveDetails bool) {
+	tableColumnLabel := []string{"ID", "Name", "Entry", "Destination", "Description"}
+	if resolveDetails {
+		tableColumnLabel = append(tableColumnLabel, "Menu (id)")
+		tableColumnLabel = append(tableColumnLabel, "Category (id)")
+	}
+	p.Keys(tableColumnLabel...)
 	for _, extension := range extensions {
-		p.Record(
+		tableRow := []string{
 			extension.ExtensionID,
 			extension.Name,
 			extension.Entry,
 			extension.Destination.ID,
-			menucolumn(extension),
 			extension.Description,
-		)
+		}
+		if resolveDetails {
+			tableRow = append(tableRow, menucolumn(extension)...)
+		}
+		p.Record(tableRow...)
 	}
 	p.Print()
 }
 
-func menucolumn(extension *extensibility.ExtensionInfo) string {
-	if extension.Menu.ID == "" {
-		return ""
+func menucolumn(extension *extensibility.ExtensionInfo) []string {
```

Comment:
this function is doing the job of two function invocations. I'd simplify the function to manage a single column and invoke it twice